build-tools/src/main/java/org/elasticsearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -42,9 +44,10 @@ public class DistributionDownloadPlugin implements Plugin<Project> {
     private static final String DOWNLOAD_REPO_NAME = "elasticsearch-downloads";
     private static final String SNAPSHOT_REPO_NAME = "elasticsearch-snapshots";
     public static final String DISTRO_EXTRACTED_CONFIG_PREFIX = "es_distro_extracted_";
+    public static final String DISTRO_CONFIG_PREFIX = "es_distro_file_";
 
     private NamedDomainObjectContainer<ElasticsearchDistribution> distributionsContainer;
-    private NamedDomainObjectContainer<DistributionResolution> distributionsResolutionStrategiesContainer;
```

Comment:
Made that simpler and also just avoiding NamedDomainObjectContainer referenced in cached configuration

Reply:
FWIW this is used in the serverless build so we should confirm it's compatible.

Reply:
I did some changes here in this PR and raised a PR in serverless that allows a smooth migration, meaning we do not break serverless when merging this PR.